### ISSUE TITLE
HRA regression when rasterizing spatial criteria

### ISSUE THREAD

**Issue by phargogh** [natcap/invest#1120]
A [user on the forums](https://community.naturalcapitalproject.org/t/habitat-risk-assessment-problems-with-risk-calculation-and-outputs/3122/15) noticed a difference in how spatial criteria were being represented between InVEST 3.9 (former HRA) and InVEST 3.12 (the new HRA implementation), specifically around how input vectors are being rasterized.

After digging into this for a while, it appears that the InVEST 3.9 implementation makes the following distinction that I missed when rewriting HRA:
* Habitat and stressor vectors are rasterized with `ALL_TOUCHED=TRUE`
* Spatial criteria vectors are rasterized with `ALL_TOUGHED=FALSE`

The current HRA does not make this distinction, but probably should, at least for backwards compatibility.

For what it's worth, I could not find this distinction in the InVEST 3.3.0 HRA.

**Comment by davemfish** [comment]
Oh, whoops, maybe I misinterpreted. My suggestion is probably _exactly_ what was ruffling the feathers in the first place. Nevermind.

**Comment by phargogh** [comment]
Yeah, the change in rasterization behavior is what was ruffling feathers in the [forums thread](https://community.naturalcapitalproject.org/t/habitat-risk-assessment-problems-with-risk-calculation-and-outputs/3122/14), and this change was by accident.

If there was a deliberate reason to have these cases be handled differently, I can't find a reference for it in the source code or the user's guide, and all the references to rasterization I can find in InVEST 3.3.0 use a consistent `ALL_TOUCHED=TRUE`, so I think it'll be worth looking into this more closely and seeing what makes more sense in the long run and then deliberately making that change.

What do you think?

**Comment by davemfish** [comment]
I'm in favor of making the rasterization internally consistent; using the same `ALL_TOUCHED` rule for all rasterizations. If a user really needs full control over the rasterization method, they have the option to do it themselves and feed rasters into the model instead of vectors, right?

My take is that the new HRA represents a correction rather than a regression.

**Comment by davemfish** [comment]
I just took a quick look at the sample data. It seems commonplace for the footprint of a spatial criteria layer to exactly match the footprint of either a habitat or stressor. So I could imagine that rasterizing with different rules could result in a problem where, after rasterizing, a pixel on the margin might have presence of both the habitat & stressor, but would be missing the spatial criteria rating associated with that habitat-stressor pair.
